vllm/entrypoints/openai/api_server.py

```diff
@@ -148,8 +165,26 @@ async def build_async_engine_client(args) -> AsyncIterator[AsyncEngineClient]:
 
 
 def mount_metrics(app: FastAPI):
-    # Add prometheus asgi middleware to route /metrics requests
-    metrics_route = Mount("/metrics", make_asgi_app())
+    # Lazy import for prometheus multiprocessing.
+    # We need to set PROMETHEUS_MULTIPROC_DIR environment variable
+    # before prometheus_client is imported.
+    # See https://prometheus.github.io/client_python/multiprocess/
+    from prometheus_client import (CollectorRegistry, make_asgi_app,
+                                   multiprocess)
+
+    prometheus_multiproc_dir_path = os.getenv("PROMETHEUS_MULTIPROC_DIR", None)
+    if prometheus_multiproc_dir_path is not None:
+        logger.info("vLLM to use %s as PROMETHEUS_MULTIPROC_DIR",
```

Comment:
Is this the value we set earlier with the temp directory or only when the user has set the variable?

Reply:
It could be either here. The only case where the env var won't be present would be when decoupled front-end is explicitly disabled i.e. it's running in single-process mode.